### PR TITLE
A new search design is implemented.

### PR DESCRIPTION
**Pull request by itopaloglu83**
It uses the allMarkers as a master list.
Every marker now has an attribute called found.
Found attribute is used to determine if that marker is found in search.
This information later used to put the found markers on the map.
The search performance is vastly improved.
However,
- showOnlyNewProperties function is now broken,
- cross browser testing is required,
- functional programming can be used instead of for loops,
- search results can be filtered by .claimed or .new before showing,
- removed functionality of showing all the markers if search yields no result,
- might introduced new bugs.

### CODE REVIEW

File: static/js/main.js [hunk 1]
```diff
@@ -96,75 +117,50 @@ $("document").ready(function() {
...
+    markers.clearLayers();
+    markers.addLayers(foundMarkers);
+
+    // Let's select the marker if there's only one of them is found
+    if (foundMarkers.length === 1 ) {
+      selectedProperty(foundMarkers[0].address, foundMarkers[0].parcelid, foundMarkers[0].claimed);
     }
   }
 
+  // NEEDS UPDATE: Broken by search update
```
Comment (DavidEBest) [inline static/js/main.js]: This should be relatively straight forward to re-implement. Places where you set the `found` flag could also take into account the state of the `new` flag.

**Comment by DavidEBest** [comment]
Seems pretty good. I'm downloading a VM so I can test on IE, but the performance seems good on Chrome and Safari.

File: static/js/main.js (hunk 1 above)
Comment (itopaloglu83) [inline static/js/main.js]: We can also find all the matching properties and then update the map honoring the new flag.

**Comment by DavidEBest** [comment]
D'oh. Didn't mean to do that...

**Comment by DavidEBest** [comment]
@itopaloglu83 Apologies! I accidentally merged this, and then reverted it since the 'Show new' functionality is broken. Would you mind re-issuing the pull request?

**Comment by itopaloglu83** [comment]
I can't. GitHub thinks that everything is up to date.

> codefordayton:master is up to date with all commits from itopaloglu83:master.

**Comment by DavidEBest** [comment]
Okay, I see my problem, https://git-scm.com/blog/2010/03/02/undoing-merges.html. Don't worry about it, I'll get it sorted out.

**Pull request merged**